script.js

```diff
@@ -9,89 +9,206 @@
-    "Emma":     64485690,
-    "Felicia":  83323669,
-    "George":   98119091,
-    "Herman":   63436894,
-};
+// 5. (Bonus) return an array where each element in this new array is the sum of the element before it and the element in its current position in the old array, e.g. the first few elements in this new array would be `[3, 12, 11, 7, 8, ...]`
 
+// var modify = function(array, n) {
 
+//     // Nothing to do when array size is 1
+//     if (n <= 1) {
+//         return;
+//     }
 
+//     // store current value of arr[0] and update it
+//     var previous = array[0]
+//     array[0] = array[0] + array[1]
 
+//     // Update rest of the array elements
+//     for (i = 1; i < n-1; i++) {
 
+//         // Store current value of next interation
+//         var current = array[i];
 
+//         // Update current value using previos value
+//         array[i] = previous + array[i+1];
 
+//         // Update previous value
+//         previous = current;
+//     }
 
+//     // Update last array element
+//     array[n-1] = previous + array[n-1]
 
+// }
 
+// modify(numArray, numArray.length);
 
+// console.log(numArray);
 
-// Exercise 5
-// ==========
-// Examine the array below and the commented out loop that was written to console.log all the names from the array, starting from the last element.
 
-var testArray = ["Alice", "Bobby", "Charles", "Daniel", "Elise", "Farnsworth", "Grace", "Horace", "Ihsan", "Jack"];
+// Exercise 4
+// ==========
+// Examine the phonebook below.
+// Write a function that takes an object, e.g. Phonebook, that uses a loop to return an object with keys and values swapped.
 
-// for (var i = 0; i < testArray.length; i++) {
-//     var name = testArray.pop();
-//     console.log(name);
+// var Phonebook = {
+//     "Aaron":    92133243,
+//     "Betty":    91120543,
+//     "Cammy":    85535657,
+//     "Denise":   64122423,
+//     "Emma":     64485690,
+//     "Felicia":  83323669,
+//     "George":   98119091,
+//     "Herman":   63436894,
 // };
 
+// var swapKeyAndValue = function(phonebook) {
+//   var newObject = {};
+//   for(var key in phonebook) {
+//     newObject[phonebook[key]] = key;
+//   }
+//   return newObject;
+// }
+
+// console.log(swapKeyAndValue(Phonebook));
+
+// Exercise 5
+// ==========
+// Examine the array below and the commented out loop that was written to console.log all the names from the array, starting from the last element.
 // Uncomment and run the code. What is wrong with the output?
 // Explain the mistake in a comment on the line/lines that you think the mistake is at.
 // Correct the code.
 
+// var testArray = ["Alice", "Bobby", "Charles", "Daniel", "Elise", "Farnsworth", "Grace", "Horace", "Ihsan", "Jack"];
 
+// for (var i = 0; i < testArray.length; i++) { // to start from the last element, should decrease i instead;
+//     arrayFromLastElement.push(testArray[i]);
+//     var name = testArray.pop(); // wrong method .pop() is used to remove last element
```

Comment:
Well, it does give the last element to the `name` variable so that it can be console logged. More specifically, the issue is that it actually alters `testArray` and so `testArray.length` keeps changing as you go through the loop. So the comparison `i < testArray.length` is not going to behave as intended.